tools/syz-execprog/execprog.go

```diff
@@ -251,27 +257,54 @@ func (ctx *Context) printHints(p *prog.Prog, info *ipc.ProgInfo) {
 	log.Logf(0, "ncomps=%v ncandidates=%v", ncomps, ncandidates)
 }
 
-func (ctx *Context) dumpCallCoverage(coverFile string, info *ipc.CallInfo) {
+func getKernelUpperBase(target *targets.Target) uint32 {
+	defaultRet := uint32(0xffffffff)
+	if target.OS == targets.Linux {
+		// Read the first 8 bytes from /proc/kallsyms.
+		f, err := os.Open("/proc/kallsyms")
+		if err != nil {
+			log.Logf(1, "could not get kernel fixup address: %v", err)
+			return defaultRet
+		}
+		defer f.Close()
+		data := make([]byte, 8)
+		_, err = f.ReadAt(data, 0)
+		if err != nil {
+			log.Logf(1, "could not get kernel fixup address: %v", err)
+			return defaultRet
+		}
+		value, err := strconv.ParseUint(string(data), 16, 32)
+		if err != nil {
+			log.Logf(1, "could not get kernel fixup address: %v", err)
+			return defaultRet
+		}
+		return uint32(value)
+	}
+	return defaultRet
+}
+
+func (ctx *Context) dumpCallCoverage(coverFile string, info *ipc.CallInfo, target *targets.Target) {
 	if len(info.Cover) == 0 {
 		return
 	}
 	buf := new(bytes.Buffer)
 	for _, pc := range info.Cover {
-		fmt.Fprintf(buf, "0x%x\n", cover.RestorePC(pc, 0xffffffff))
+		upperBase := getKernelUpperBase(target)
```

Comment:
I think we need to do it just once per the whole `syz-execprog` execution.

Reply:
Done, ptal